components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -195,7 +197,11 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                 context.initializeAnalyticsData();
             } else {
                 returning = true;
-                context = FrameworkUtils.getContextData(request);
+                if (FrameworkUtils.isAuthenticationContextExpiryEnabled()) {
```

Comment:
We can move this check inside Framework util getAuthenticationContextFromCache method and handle internally

Reply:
But do we need this config at all ?

Reply:
Added this config since we modifying the behavior and error message. If someone want to stay with the old behavior, they can do so by disabling this config

Reply:
Fixed